### PR TITLE
Hand Landmarker C Tasks API and tests

### PR DESCRIPTION
**Pull request by kinaryml**
- Includes minor typo fixes in the C Gesture Recognizer tasks API

### CODE REVIEW

File: mediapipe/tasks/c/vision/hand_landmarker/hand_landmarker.cc [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+      static_cast<ImageFormat::Format>(image.image_frame.format),
+      image.image_frame.image_buffer, image.image_frame.width,
+      image.image_frame.height);
+
+  if (!img.ok()) {
+    ABSL_LOG(ERROR) << "Failed to create Image: " << img.status();
+    return CppProcessError(img.status(), error_msg);
+  }
+
+  auto cpp_detector = static_cast<HandLandmarker*>(detector);
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/c/vision/hand_landmarker/hand_landmarker.cc:154]: s/detector/landmarker

File: mediapipe/tasks/c/vision/hand_landmarker/hand_landmarker.cc [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/c/vision/hand_landmarker/hand_landmarker.cc:216]: s/detector/landmarker

File: mediapipe/tasks/c/vision/hand_landmarker/hand_landmarker.h [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+// Creates an HandLandmarker from the provided `options`.
+// Returns a pointer to the hand landmarker on success.
+// If an error occurs, returns `nullptr` and sets the error parameter to an
+// an error message (if `error_msg` is not `nullptr`). You must free the memory
+// allocated for the error message.
+MP_EXPORT void* hand_landmarker_create(struct HandLandmarkerOptions* options,
+                                       char** error_msg);
+
+// Performs gesture recognition on the input `image`. Returns `0` on success.
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/c/vision/hand_landmarker/hand_landmarker.h:85]: hand landmark detection

File: mediapipe/tasks/c/vision/hand_landmarker/hand_landmarker.h [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+// If an error occurs, returns an error code and sets the error parameter to an
+// an error message (if `error_msg` is not `nullptr`). You must free the memory
+// allocated for the error message.
+MP_EXPORT int hand_landmarker_detect_for_video(void* detector,
+                                               const MpImage& image,
+                                               int64_t timestamp_ms,
+                                               HandLandmarkerResult* result,
+                                               char** error_msg);
+
+// Sends live image data to gesture recognition, and the results will be
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/c/vision/hand_landmarker/hand_landmarker.h:108]: the hand landmark detector

File: mediapipe/tasks/c/vision/hand_landmarker/hand_landmarker_test.cc [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+                           /* model_asset_buffer_count= */ 0,
+                           /* model_asset_path= */ model_path.c_str()},
+      /* running_mode= */ RunningMode::IMAGE,
+      /* num_hands= */ 1,
+      /* min_hand_detection_confidence= */ 0.5,
+      /* min_hand_presence_confidence= */ 0.5,
+      /* min_tracking_confidence= */ 0.5,
+  };
+
+  void* detector = hand_landmarker_create(&options, /* error_msg */ nullptr);
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/c/vision/hand_landmarker/hand_landmarker_test.cc:93]: landmarker 

(here and everywhere)

File: mediapipe/tasks/c/vision/hand_landmarker/hand_landmarker_test.cc [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+}
+
+// A structure to support LiveStreamModeTest below. This structure holds a
+// static method `Fn` for a callback function of C API. A `static` qualifier
+// allows to take an address of the method to follow API style. Another static
+// struct member is `last_timestamp` that is used to verify that current
+// timestamp is greater than the previous one.
+struct LiveStreamModeCallback {
+  static int64_t last_timestamp;
+  static void Fn(HandLandmarkerResult* detector_result, const MpImage& image,
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/c/vision/hand_landmarker/hand_landmarker_test.cc:157]: landmarker_result

File: mediapipe/tasks/c/vision/hand_landmarker/hand_landmarker_test.cc [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+                 int64_t timestamp, char* error_msg) {
+    ASSERT_NE(detector_result, nullptr);
+    ASSERT_EQ(error_msg, nullptr);
+    MatchesHandLandmarkerResult(detector_result, kScorePrecision,
+                                kLandmarkPrecision);
+    EXPECT_GT(image.image_frame.width, 0);
+    EXPECT_GT(image.image_frame.height, 0);
+    EXPECT_GT(timestamp, last_timestamp);
+    last_timestamp++;
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/c/vision/hand_landmarker/hand_landmarker_test.cc:166]: Prefer ++last_timestamp (per Google styleguide)

**Comment by kinaryml** [comment]
@schmidt-sebastian I've resolved the nits concerning naming now. PTAL.

**Pull request merged**